waspc/src/Wasp/Generator/ServerGenerator/JobGenerator.hs

```diff
@@ -112,22 +118,20 @@ genJobExecutors = return $ jobExecutorFds ++ jobExecutorHelperFds
 
     jobExecutorHelperFds :: [FileDraft]
     jobExecutorHelperFds =
-      [ C.mkTmplFd $ jobsDirInServerTemplatesDir SP.</> [relfile|core/pgBoss/pgBoss.js|],
-        C.mkTmplFd $ jobsDirInServerTemplatesDir SP.</> [relfile|core/Job.js|],
-        C.mkTmplFd $ jobsDirInServerTemplatesDir SP.</> [relfile|core/SubmittedJob.js|]
+      [ C.mkTmplFd $ jobsDirInServerTemplatesDir SP.</> [relfile|core/pgBoss/pgBoss.ts|],
+        C.mkTmplFd $ jobsDirInServerTemplatesDir SP.</> [relfile|core/job.ts|]
       ]
 
+    executorJobTemplateInServerTemplatesDir :: JobExecutor -> Path SP.System (Rel ServerTemplatesDir) File'
+    executorJobTemplateInServerTemplatesDir = (jobsDirInServerTemplatesDir SP.</>) . executorJobTemplateInJobsDir "ts"
+
 data JobsDir
 
 jobsDirInServerTemplatesDir :: Path' (Rel ServerTemplatesDir) (Dir JobsDir)
 jobsDirInServerTemplatesDir = srcDirInServerTemplatesDir SP.</> [reldir|jobs|]
 
-executorJobTemplateInServerTemplatesDir :: JobExecutor -> Path SP.System (Rel ServerTemplatesDir) File'
-executorJobTemplateInServerTemplatesDir = (jobsDirInServerTemplatesDir SP.</>) . executorJobTemplateInJobsDir
-
-executorJobTemplateInJobsDir :: JobExecutor -> Path' (Rel JobsDir) File'
-executorJobTemplateInJobsDir PgBoss = [relfile|core/pgBoss/pgBossJob.js|]
-executorJobTemplateInJobsDir Simple = [relfile|core/simpleJob.js|]
+executorJobTemplateInJobsDir :: String -> JobExecutor -> Path' (Rel JobsDir) File'
+executorJobTemplateInJobsDir ext PgBoss = fromJust $ SP.parseRelFile $ "core/pgBoss/pgBossJob" <> "." <> ext
```

Comment:
Why aren't we using `++`? It's fine either way, just wondering.

Reply:
Flexing Haskell knowledge? I'd leave it like this if it's fine with you